server/github/builder.go

```diff
@@ -34,3 +37,40 @@ func ImageFromTemplate(t *template.Template) ImageBuilder {
 		return image.Decode(buf.String())
 	})
 }
+
+// conveyorClient mocks the interface to the Conveyor API.
+type conveyorClient interface {
+	Build(io.Writer, conveyor.BuildCreateOpts) (*conveyor.Artifact, error)
+}
+
+// ConveyorImageBuilder provides an ImageBuilder implementation that
+// integrations with the Conveyor (https://github.com/remind101/conveyor) Docker
+// build system. If enabled, Empire will check if an Artifact in Conveyor exists
+// for the git commit, and will trigger Conveyor to build it if it doesn't
+// exist.
+type ConveyorImageBuilder struct {
+	client conveyorClient
+}
+
+// NewConveyorImageBuilder returns a new ConveyorImageBuilder implementation
+// that uses the given client.
+func NewConveyorImageBuilder(c *conveyor.Service) *ConveyorImageBuilder {
+	return &ConveyorImageBuilder{
+		client: c,
+	}
+}
+
+func (c *ConveyorImageBuilder) BuildImage(ctx context.Context, event events.Deployment) (image.Image, error) {
+	w := ioutil.Discard // TODO: Allow ImageBuilder implementations to write to the deployment logs?
+
+	a, err := c.client.Build(w, conveyor.BuildCreateOpts{
+		Repository: event.Repository.FullName,
+		Branch:     "", // TODO,
```

Comment:
Actually, I think we should not set Branch at all since we really have no idea what the ref was. In most cases, Conveyor will have already started the build, using the branch cache and Empire will just wait for it to complete.